src/presentation/http/http-api.ts

```diff
@@ -0,0 +1,234 @@
+   * @param config - http server config
+   */
+  private static async addCookies(server: FastifyServer, config: HttpApiConfig): Promise<void> {
+    await server.register(cookie, {
+      secret: config.cookieSecret,
+    });
+  }
+
+  /**
+   * Registers all routers
+   *
+   * @param server - fastify server instance
+   * @param config - http server config
+   * @param domainServices - instances of domain services
+   * @param middlewares - middlewares
+   */
+  private static async addAPI(server: FastifyServer, config: HttpApiConfig, domainServices: DomainServices, middlewares: Middlewares): Promise<void> {
+    await server.register(NoteRouter, {
+      prefix: '/note',
+      noteService: domainServices.noteService,
+      middlewares: middlewares,
+    });
+
+    await server.register(OauthRouter, {
+      prefix: '/oauth',
+      userService: domainServices.userService,
+      authService: domainServices.authService,
+      cookieDomain: config.cookieDomain,
+    });
+
+    await server.register(AuthRouter, {
+      prefix: '/auth',
+      authService: domainServices.authService,
+    });
+
+    await server.register(UserRouter, {
+      prefix: '/user',
+      userService: domainServices.userService,
+      middlewares: middlewares,
+    });
+
+    await server.register(AIRouter, {
+      prefix: '/ai',
+      aiService: domainServices.aiService,
+    });
+
+    await server.register(EditorToolsRouter, {
+      prefix: '/editor-tools',
+      editorToolsService: domainServices.editorToolsService,
+    });
+  }
+
+  /**
+   * Registers oauth2 plugin
+   *
+   * @param server - fastify server instance
+   * @param config - http server config
+   */
+  private static async addOauth2(server: FastifyServer, config: HttpApiConfig): Promise<void> {
+    await server.register(fastifyOauth2, {
+      name: 'googleOAuth2',
+      scope: ['profile', 'email'],
+      credentials: {
+        client: {
+          id: config.oauth2.google.clientId,
+          secret: config.oauth2.google.clientSecret,
+        },
+        auth: fastifyOauth2.GOOGLE_CONFIGURATION,
+      },
+      startRedirectPath: config.oauth2.google.redirectUrl,
+      callbackUri: config.oauth2.google.callbackUrl,
+    });
+  }
+
+  /**
+   * Allows cors for allowed origins from config
+   *
+   * @param server - fastify server instance
+   * @param config - http server config
+   */
+  private static async addCORS(server: FastifyServer, config: HttpApiConfig): Promise<void> {
+    await server.register(cors, {
+      origin: config.allowedOrigins,
+    });
+  }
+
+  /**
+   * Add Fastify schema for validation and serialization
+   *
+   * @param server - fastify server instance
+   */
+  private static addSchema(server: FastifyServer): void {
+    server.addSchema(UserSchema);
+  }
+
+  /**
+   * Custom method for sending 404 error
+   *
+   * @param server - fastify server instance
+   */
+  private static add404Handling(server: FastifyServer): void {
```

Comment:
maybe `addDecorators`? There will be several more added soon

Reply:
➕